src/commands/pressable-grant-access.php

```diff
@@ -2,91 +2,362 @@
-				exit;
-			}
+		// If we still don't have en email fail.
+		if ( empty( $email ) ) {
+			$output->writeln( '<error>You must supply a valid email address.</error>' );
+			exit;
 		}
 
-		$output->writeln( '<comment>Granting ' . $email . ' access to site ' . $site . '.</comment>' );
+		return $email;
+	}
 
-		$site_id = null;
-		if ( is_numeric( $site ) ) {
-			$site_id = $site;
-		} else {
-			$search_site = $this->api_helper->call_pressable_api( sprintf('sites', $site), 'GET', array() );
-			$output->writeln( '<comment>Looping through ' . count($search_site->data) . ' domains searching for ' . $site . '.</comment>' );
-
-			// Loop to find site_id by domain
-			foreach ( $search_site->data as $key => $val ) {
-				if ( $val->url === $site) {
-					$site_id = $val->id;
-					$output->writeln( "<comment>Found it! It's site ID {$site_id}</comment>" );
-					break;
-				}
+	/**
+	 * Gets the search term either form Input or prompted for.
+	 *
+	 * @param \Symfony\Component\Console\Input\InputInterface $input
+	 * @param \Symfony\Component\Console\Output\OutputInterface $output
+	 * @return string
+	 */
+	private function get_search_term( InputInterface $input, OutputInterface $output ): string {
+		// Attempt to get search term from Input.
+		$search_term = $input->getOption( 'search' );
+
+		// If we don't have a search term, prompt the user.
+		if ( empty( $search_term ) ) {
+			$search_term = $this->get_question_helper()->ask( $input, $output, $this->ask_for_search_term() );
+		}
+
+		// If we still don't have a search term fail.
+		if ( empty( $search_term ) ) {
+			$output->writeln( '<error>You must supply a valid search term to look for matching domains.</error>' );
+			exit;
+		}
+
+		return $search_term;
+	}
+
+	/**
+	 * Gets the site url either form Input or prompted for.
+	 *
+	 * @param \Symfony\Component\Console\Input\InputInterface $input
+	 * @param \Symfony\Component\Console\Output\OutputInterface $output
+	 * @return string
+	 */
+	private function get_site( InputInterface $input, OutputInterface $output ): string {
+		// Attempt to get site name from Input.
+		$site_name = $input->getOption( 'site' );
+
+		// If we don't have a site name, prompt the user.
+		if ( empty( $site_name ) ) {
+			$site_name = $this->get_question_helper()->ask( $input, $output, $this->ask_site_name() );
+		}
+
+		// If we still don't have a site name fail.
+		if ( empty( $site_name ) ) {
+			$output->writeln( '<error>You must supply a valid site URL.</error>' );
+			exit;
+		}
+
+		return $site_name;
+	}
+
+	/*************************************
+	 *************************************
+	 *             QUESTIONS             *
+	 *************************************
+	 *************************************/
+
+
+	/**
+	 * Returns the question for looking for a site.
+	 *
+	 * @return \Symfony\Component\Console\Question\Question
+	 */
+	private function ask_site_name(): Question {
+		return new Question( 'Please enter the site?', false );
```

Comment:
Can you add an empty space at the end of each question? That way, when the user types in there's a space between the question and their answer.

Also, minor thing, but I don't think we need the `?` at the end of the sentence, because it's not an actual question, maybe change it with a `:` ?

eg: 'Please enter the site: '
or, 'What's the site name? '

Reply:
`?` have been replaced with `:` and trailing whitespace has been added